src/terminal_renderer/TextRenderer.cpp

```diff
@@ -251,12 +251,12 @@ optional<TextRenderer::DataRef> TextRenderer::getTextureInfo(text::glyph_key con
             if (optional<DataRef> const dataRef = ta->get(_id); dataRef.has_value())
                 return dataRef;
 
-    bool const colored = textShaper_.has_color(_id.font);
-
     auto theGlyphOpt = textShaper_.rasterize(_id, fontDescriptions_.renderMode);
     if (!theGlyphOpt.has_value())
         return nullopt;
 
+    bool const colored = textShaper_.has_color(_id.font);
```

Comment:
This is a bit awkward, because DWrite engine needs to first `rasterize` then have the `has_color` available.

Reply:
has_color is an architectural work around that i have to really address and fix in the upper layer. I am sorry for this inconvenience. I will try to get rid of the need of this member function ASAP (maybe tonight).